valhalla/mjolnir/landmark_database_builder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <boost/property_tree/ptree.hpp>
+#include <cstdint>
+#include <iostream>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+enum class AccessMode { ReadWriteCreate, ReadOnly, ReadWrite };
+
+struct Landmark {
+  std::string name;
+  std::string type;
+  double lng;
+  double lat;
+};
+
+struct LandmarkDatabase {
+public:
+  LandmarkDatabase(const std::string& db_name, AccessMode access_mode = AccessMode::ReadOnly)
+      : db(nullptr), database(db_name), access_mode_(access_mode) {
+    if (!connect_database()) {
+      LOG_ERROR("cannot connect to database");
+    }
+  }
+  ~LandmarkDatabase() {
+    if (did_inserts && !vacuum_analyze()) {
+      LOG_ERROR("cannot do vacuum and analyze");
+    }
+    close_database();
+  }
+
+  bool insert_landmark(const Landmark& landmark);
```

Comment:
i know it seems convenient but it is wasteful for the parser to have to allocate (copy) the parsed data into a new structure just so you can push it into the database. if the builder itself takes a long time this is a place we can look to speed it up. for now we can leave it alone